tests/Doctrine/Tests/ORM/Query/SelectSqlGenerationTest.php

```diff
@@ -2242,6 +2259,43 @@ public function testHavingSupportResultVariableInAggregateFunction()
             'SELECT COUNT(c0_.name) AS sclr_0 FROM cms_users c0_ HAVING sclr_0 IS NULL'
         );
     }
+
+    /**
+     * @group DDC-3701
+     */
+    public function testFROMClauseAcceptsFQCNWithLeadingBackslash()
+    {
+        $this->_em->createQuery(
+            'SELECT c FROM \Doctrine\Tests\Models\Company\CompanyContract c'
```

Comment:
This is un-wanted. In DQL, you always use FQCNs, therefore the leading backslash should never match.